### PR TITLE
Stake referral metric

### PR DESCRIPTION
**Pull request by Jeday**
### Description
Adds metric indicator for stake tx on widget. 

### Testing notes

- send stake transaction with referral or without
- check raw tx call data on etherscan
- tx must succeced, no error
- you should see `01` at the end of data after ref address(or zero address)

Example:
image

### Checklist:

- [x] Checked the changes locally.
- [ ] Created / updated analytics events.
- [ ] Created / updated the technical documentation (README.md / [docs](https://docs.lido.fi/) / etc.).
- [ ] Affects / requires changes in other services (Matomo / Sentry / CloudFlare / etc.).

### CODE REVIEW

**Review by TheDZhon — APPROVED** [review]
What a wonderful idea you came up with :+1:  

Approving the _approach_ from my side.

---

NB: the presented implementation itself is still the subject of a proper review according to the established process.

File: features/stake/stake-form/utils.ts [hunk 1]
```diff
@@ -13,16 +14,34 @@ export const getAddress = async (
   input: string,
   providerRpc: BaseProvider,
 ): Promise<string> => {
+  let address;
   try {
-    if (isAddress(input)) return input;
-    const address = await providerRpc.resolveName(input);
-    if (address) return address;
+    // extract address from url or from ens
+    if (isAddress(input)) {
```
Comment (DiRaiks) [inline features/stake/stake-form/utils.ts:20]: Let's revert the previous code. I don't see the point in let

File: features/stake/stake-form/utils.ts [hunk 2]
```diff
@@ -13,16 +14,34 @@ export const getAddress = async (
...
   }
+  // if code gets here, ref is invalid and we need to throw error
   throw new ReferralAddressError();
 };
 
+// adds metrics indicator for widget tx
+export const applyCalldataSuffix = (tx: PopulatedTransaction) => {
+  if (!config.ipfsMode) {
+    invariant(tx.data, 'transaction must have calldata');
+    tx.data = tx.data + '01';
```
Comment (DiRaiks) [inline features/stake/stake-form/utils.ts:40]: lets move 01 to consts

**Review by DiRaiks — APPROVED** [review]
lgtm

**Pull request merged**